### DISCUSSION TITLE
Where clause entered incorrectly

### DISCUSSION THREAD

**Discussion by randybarton24** [jamessimone/apex-rollup#471] [Q&A]
@jamessimone thought I'd open this as a separate discussion, since it's unrelated to the other issue we've been discussing.  I've migrated some DLRS rollups but I'm getting the error message below from Apex Rollup.  The where clause in the error message doesn't match what's saved in the metadata record.  I've changed it to be `= '` instead of `IN ('` and that seems to fix it, but just thought I'd give you a heads up.  I think we originally used `IN ('` because we had multiple values to check against, but later removed them until getting down to one value and left the IN instead of changing to =

Error Message:
System.IllegalArgumentException: Where clause entered incorrectly: Product_Family__c = ('Campaign Registration and Administration Fees') AND RecurSoft__Billing_Status__c = ('Regular Invoicing') AND RecurSoft__Billing_Frequency__c != 'One-Time' Exception: Ending position out of bounds: -1

Actual Where Clause:
Product_Family__c IN ('Campaign Registration and Administration Fees') AND RecurSoft__Billing_Status__c IN ('Regular Invoicing') AND RecurSoft__Billing_Frequency__c != 'One-Time'

**Comment by jamessimone** [comment]
I'll take a look at this one after I release 1.5.77, IN is definitely supported but it could be an edge case with the value you're supplying having "and" in the actual value. That was an issue a _really_ long time ago, and I have test coverage for it, so it could be something else as well, but thanks for providing the actual clause you used - that should make it relatively easy to reproduce whatever the actual issue is and fix that up for you.
